packages/lib/src/interactions/box.ts

```diff
@@ -0,0 +1,61 @@
+import { Box3, Vector3 } from 'three';
+
+import type { Size } from '../vis/models';
+
+const ZERO_VECTOR = new Vector3(0, 0, 0);
+
+class Box extends Box3 {
+  public get size(): Size {
+    const { x: width, y: height } = this.getSize(new Vector3());
+    return { width, height };
```

Comment:
`getSize` returns a vector. I'm still a little unsure as to whether this is a good thing or not. Destructuring a size vector with `{ x: width, y: height }` or creating one with `new Vector3(width, height, 0)` definitely doesn't feel right, but a vector is useful for working with methods like `expandByVector` or for making operations on it. For now, I've decided to stick with our `Size` data structure and I've also added an `expandBySize` method to abstract away `expandByVector`.